### ISSUE TITLE
I did permission set at security and privacy but  at Log tap Permission denied message occur

### ISSUE THREAD

**Issue by JongHoooon** [pqrs-org/Karabiner-Elements#3267]
I get m1 pro Macbook Pro yesterday and I updated to Vemtura 

then I installed Karabiner and at system setting - security and privacy I checked permission  2 - 3 times 

Log tap releases Permission denied message occur 

[2022-10-25 16:22:01.962] [error] [console_user_server] grabber_client connect_failed: Permission denied
[2022-10-25 16:22:03.035] [error] [console_user_server] grabber_client connect_failed: Permission denied
[2022-10-25 16:22:04.106] [error] [console_user_server] grabber_client connect_failed: Permission denied
[2022-10-25 16:22:05.149] [error] [console_user_server] grabber_client connect_failed: Permission denied
[2022-10-25 16:22:28.642] [info] [grabber] version 14.10.0
[2022-10-25 16:22:28.674] [info] [grabber] device_open_permitted
[2022-10-25 16:22:28.676] [info] [grabber] task_policy_set is called.
[2022-10-25 16:22:28.677] [info] [grabber] session_monitor_receiver is initialized
[2022-10-25 16:22:28.684] [info] [grabber] receiver is initialized
[2022-10-25 16:22:28.684] [info] [grabber] session_monitor_receiver: bound
[2022-10-25 16:22:28.685] [info] [grabber] receiver: bound
[2022-10-25 16:22:28.685] [info] [grabber] receiver: chown socket: 0
[2022-10-25 16:22:28.730] [info] [observer] version 14.10.0
[2022-10-25 16:22:28.737] [info] [virtual_hid_device_service] version 1.30.0
[2022-10-25 16:22:28.737] [info] [virtual_hid_device_service] io_service_client::async_start
[2022-10-25 16:22:28.737] [info] [virtual_hid_device_service] virtual_hid_device_service_server is initialized
[2022-10-25 16:22:28.738] [info] [virtual_hid_device_service] virtual_hid_device_service_server: bound
[2022-10-25 16:22:28.740] [info] [virtual_hid_device_service] driver_version_ is changed: 10600
[2022-10-25 16:22:28.740] [info] [virtual_hid_device_service] io_service_client::opened
[2022-10-25 16:22:28.772] [info] [observer] device_open_permitted
[2022-10-25 16:22:28.775] [info] [observer] device_observer is started.
[2022-10-25 16:22:28.776] [info] [observer] grabber_client is started.
[2022-10-25 16:22:28.777] [info] [observer] matching device:
[2022-10-25 16:22:28.777] [info] [observer]   registry_entry_id: 4294969569
[2022-10-25 16:22:28.777] [info] [observer]   manufacturer: Apple Inc.
[2022-10-25 16:22:28.777] [info] [observer]   product: Apple Internal Keyboard / Trackpad
[2022-10-25 16:22:28.777] [info] [observer]   vendor_id: 1452
[2022-10-25 16:22:28.777] [info] [observer]   product_id: 834
[2022-10-25 16:22:28.777] [info] [observer]   location_id: 0x2f
[2022-10-25 16:22:28.777] [info] [observer]   serial_number: FM72014018JPDY9AG+EBT
[2022-10-25 16:22:28.778] [info] [observer]   is_keyboard: true
[2022-10-25 16:22:28.779] [info] [observer]   is_pointing_device: false
[2022-10-25 16:22:28.779] [info] [observer] rescan devices...
[2022-10-25 16:22:28.779] [info] [observer] grabber_client is connected.
[2022-10-25 16:22:28.779] [info] [observer] rescan devices...
[2022-10-25 16:22:28.779] [info] [observer] matching device:
[2022-10-25 16:22:28.779] [info] [observer]   registry_entry_id: 4294969571
[2022-10-25 16:22:28.779] [info] [observer]   manufacturer: Apple Inc.
[2022-10-25 16:22:28.779] [info] [observer]   product: Apple Internal Keyboard / Trackpad
[2022-10-25 16:22:28.779] [info] [observer]   vendor_id: 1452
[2022-10-25 16:22:28.779] [info] [observer]   product_id: 834
[2022-10-25 16:22:28.779] [info] [observer]   location_id: 0x2f
[2022-10-25 16:22:28.779] [info] [observer]   serial_number: FM72014018JPDY9AG+EBT
[2022-10-25 16:22:28.780] [info] [observer]   is_keyboard: false
[2022-10-25 16:22:28.780] [info] [observer]   is_pointing_device: true
[2022-10-25 16:22:28.780] [info] [observer] rescan devices...
[2022-10-25 16:22:28.780] [info] [observer] Apple Internal Keyboard / Trackpad (device_id:4294969569) is observed.
[2022-10-25 16:22:28.780] [info] [observer] Apple Internal Keyboard / Trackpad (device_id:4294969571) is observed.
[2022-10-25 16:23:51.565] [info] [session_monitor] version 14.10.0
[2022-10-25 16:23:51.567] [info] [console_user_server] version 14.10.0
[2022-10-25 16:23:51.585] [info] [session_monitor] session_monitor_receiver_client is started.
[2022-10-25 16:23:51.588] [info] [session_monitor] on_console_changed: true
[2022-10-25 16:23:51.589] [info] [grabber] current_console_user_id: 501
[2022-10-25 16:23:51.589] [info] [grabber] receiver is terminated
[2022-10-25 16:23:51.589] [info] [session_monitor] session_monitor_receiver_client is connected.
[2022-10-25 16:23:51.589] [info] [session_monitor] connected
[2022-10-25 16:23:51.590] [info] [grabber] receiver is initialized
[2022-10-25 16:23:51.590] [info] [grabber] receiver: bound
[2022-10-25 16:23:51.590] [info] [grabber] receiver: chown socket: 501
[2022-10-25 16:23:51.594] [error] [observer] grabber_client error: Connection reset by peer
[2022-10-25 16:23:51.594] [info] [observer] grabber_client is closed.
[2022-10-25 16:23:52.601] [info] [observer] grabber_client is connected.
[2022-10-25 16:23:52.601] [info] [observer] rescan devices...
[2022-10-25 16:23:58.190] [info] [console_user_server] on_console_changed: on_console:true
[2022-10-25 16:23:58.192] [info] [console_user_server] grabber_client is started.
[2022-10-25 16:23:58.193] [error] [console_user_server] grabber_client connect_failed: Permission denied
[2022-10-25 16:23:59.223] [error] [console_user_server] grabber_client connect_failed: Permission denied
[2022-10-25 16:24:00.292] [error] [console_user_server] grabber_client connect_failed: Permission denied

this is my log 

I used karabiner very usefully please help me

**Comment by tekezo** [comment]
Could you open Terminal.app and execute the following command?

```shell
find '/Library/Application Support/org.pqrs/tmp' -ls
```

The expected result:
```
35833748        0 drwxr-xr-x   12 root             wheel                 384 Oct 25 18:45 /Library/Application Support/org.pqrs/tmp
35833749        0 drwx------    6 root             wheel                 192 Oct 25 04:36 /Library/Application Support/org.pqrs/tmp/rootonly
43104448       16 -rw-r--r--    1 root             wheel                5924 Oct 25 18:45 /Library/Application Support/org.pqrs/tmp/karabiner_grabber_manipulator_environment.json
35849963        0 drwxr-xr-x    3 root             wheel                  96 Sep 17 13:50 /Library/Application Support/org.pqrs/tmp/user
35849964        0 drwx------    3 tekezo           wheel                  96 Oct 25 04:36 /Library/Application Support/org.pqrs/tmp/user/501
42931942        0 drwxr-xr-x    3 tekezo           wheel                  96 Oct 25 04:36 /Library/Application Support/org.pqrs/tmp/user/501/grabber_client
42931943        0 srwxr-xr-x    1 tekezo           wheel                   0 Oct 25 04:36 /Library/Application Support/org.pqrs/tmp/user/501/grabber_client/17211871acc722e8.sock
40014021        8 -rw-r--r--    1 root             wheel                 104 Oct  9 09:21 /Library/Application Support/org.pqrs/tmp/karabiner_grabber_state.json
42738382        0 drwxr-xr-x    3 root             wheel                  96 Oct 24 08:26 /Library/Application Support/org.pqrs/tmp/krbn_grabber
42738383        0 srw-------    1 tekezo           wheel                   0 Oct 24 08:26 /Library/Application Support/org.pqrs/tmp/krbn_grabber/1720d66a475f5518.sock
1729952        0 drwxr-xr-x    5 root             wheel                 160 Mar 19  2022 /Library/Application Support/org.pqrs/tmp/pid
1730891        8 -rw-r--r--    1 root             wheel                   4 Oct  9 09:20 /Library/Application Support/org.pqrs/tmp/pid/karabiner_session_monitor.501.pid
1729953        8 -rw-r--r--    1 root             wheel                   4 Oct  9 09:20 /Library/Application Support/org.pqrs/tmp/pid/karabiner_grabber.pid
1729958        8 -rw-r--r--    1 root             wheel                   4 Oct  9 09:20 /Library/Application Support/org.pqrs/tmp/pid/karabiner_observer.pid
43100774        8 -rw-r--r--    1 root             wheel                2208 Oct 25 18:14 /Library/Application Support/org.pqrs/tmp/karabiner_grabber_devices.json
43100775        8 -rw-r--r--    1 root             wheel                1728 Oct 25 18:14 /Library/Application Support/org.pqrs/tmp/karabiner_grabber_device_details.json
40012852        8 -rw-r--r--    1 root             wheel                  41 Oct  9 09:20 /Library/Application Support/org.pqrs/tmp/karabiner_observer_state.json
43087337        8 -rw-r--r--    1 root             wheel                  18 Oct 25 16:34 /Library/Application Support/org.pqrs/tmp/karabiner_notification_message.json
```

**Comment by rhty** [comment]
JFYI

I had the same symptom on my M1 Pro MacBook Pro and got the following result by executing the command.

```shell
> find '/Library/Application Support/org.pqrs/tmp' -ls
3759780        0 drwxr-xr-x    6 root             wheel                 192 10 27 05:47 /Library/Application Support/org.pqrs/tmp
3759815        0 drwx------    4 root             wheel                 128 10 26 16:35 /Library/Application Support/org.pqrs/tmp/rootonly
find: /Library/Application Support/org.pqrs/tmp/rootonly: Permission denied
3759781        0 drwxr-xr-x    5 root             wheel                 160 10 26 16:35 /Library/Application Support/org.pqrs/tmp/pid
3759782        8 -rw-r--r--    1 root             wheel                   6 10 27 05:47 /Library/Application Support/org.pqrs/tmp/pid/karabiner_grabber.pid
3774758        8 -rw-r--r--    1 root             wheel                   6 10 26 16:35 /Library/Application Support/org.pqrs/tmp/pid/karabiner_session_monitor.503.pid
3759787        8 -rw-r--r--    1 root             wheel                   6 10 26 16:36 /Library/Application Support/org.pqrs/tmp/pid/karabiner_observer.pid
3922193        8 -rw-r--r--    1 root             wheel                  42 10 27 05:47 /Library/Application Support/org.pqrs/tmp/karabiner_grabber_state.json.tmp
3775238        8 -rw-r--r--    1 root             wheel                  41 10 26 16:36 /Library/Application Support/org.pqrs/tmp/karabiner_observer_state.json
```

After I re-installed Karabiner, it worked and got the nearly same result as expected only with some different random numbers as a result of the command.

**Comment by tekezo** [comment]
Hmm, the `/Library/Application Support/org.pqrs/tmp/user` directory, which is necessary to work,  is not created.

Does restarting macOS resolve the issue?
Otherwise does third vendor antivirus software block Karabiner-Elements from the file write?

**Comment by PawelLipski** [comment]
Huh +1, I've got the same...

```
[2023-01-25 18:23:11.404] [info] [virtual_hid_device_service] version 2.0.0
[2023-01-25 18:23:11.509] [info] [virtual_hid_device_service] register Karabiner-DriverKit-VirtualHIDDeviceClient.app: noErr
[2023-01-25 18:23:11.614] [info] [grabber] version 14.11.0
[2023-01-25 18:23:11.654] [info] [grabber] launch_services::register_application /Applications/Karabiner-Elements.app: noErr
[2023-01-25 18:23:11.658] [info] [grabber] Karabiner-Elements.app path: file:///Applications/Karabiner-Elements.app/
[2023-01-25 18:23:11.739] [info] [virtual_hid_device_service] virtual_hid_device_service_server is initialized
[2023-01-25 18:23:11.740] [info] [virtual_hid_device_service] virtual_hid_device_service_server: bound
[2023-01-25 18:23:11.842] [info] [observer] version 14.11.0
[2023-01-25 18:23:13.198] [info] [grabber] device_open_permitted
[2023-01-25 18:23:13.202] [info] [grabber] task_policy_set is called.
[2023-01-25 18:23:13.205] [info] [grabber] session_monitor_receiver is initialized
[2023-01-25 18:23:13.215] [info] [observer] device_open_permitted
[2023-01-25 18:23:13.227] [info] [observer] device_observer is started.
[2023-01-25 18:23:13.235] [info] [observer] grabber_client is started.
[2023-01-25 18:23:13.238] [info] [observer] matching device:
[2023-01-25 18:23:13.238] [info] [observer]   registry_entry_id: <REDACTED>
[2023-01-25 18:23:13.238] [info] [observer]   manufacturer: Apple Inc.
[2023-01-25 18:23:13.238] [info] [observer]   product: Apple Internal Keyboard / Trackpad
[2023-01-25 18:23:13.238] [info] [observer]   vendor_id: 1452
[2023-01-25 18:23:13.239] [info] [observer]   product_id: 832
[2023-01-25 18:23:13.239] [info] [observer]   location_id: <REDACTED>
[2023-01-25 18:23:13.239] [info] [observer]   serial_number: <REDACTED>
[2023-01-25 18:23:13.242] [info] [observer]   is_keyboard: true
[2023-01-25 18:23:13.245] [info] [observer]   is_pointing_device: false
[2023-01-25 18:23:13.246] [info] [observer] rescan devices...
[2023-01-25 18:23:13.246] [error] [observer] grabber_client connect_failed: Connection refused
[2023-01-25 18:23:13.246] [info] [observer] matching device:
[2023-01-25 18:23:13.246] [info] [observer]   registry_entry_id: <REDACTED>
[2023-01-25 18:23:13.246] [info] [observer]   manufacturer: Apple Inc.
[2023-01-25 18:23:13.246] [info] [observer]   product: Apple Internal Keyboard / Trackpad
[2023-01-25 18:23:13.246] [info] [observer]   vendor_id: 1452
[2023-01-25 18:23:13.246] [info] [observer]   product_id: 832
[2023-01-25 18:23:13.246] [info] [observer]   location_id: <REDACTED>
[2023-01-25 18:23:13.246] [info] [observer]   serial_number: <REDACTED>
[2023-01-25 18:23:13.260] [info] [observer]   is_keyboard: false
[2023-01-25 18:23:13.271] [info] [observer]   is_pointing_device: true
[2023-01-25 18:23:13.271] [info] [observer] Apple Internal Keyboard / Trackpad (device_id:<REDACTED>) is observed.
[2023-01-25 18:23:13.271] [info] [observer] rescan devices...
[2023-01-25 18:23:13.281] [info] [observer] Apple Internal Keyboard / Trackpad (device_id:<REDACTED>) is observed.
[2023-01-25 18:23:14.247] [error] [observer] grabber_client connect_failed: Connection refused
[2023-01-25 18:23:14.523] [info] [grabber] receiver is initialized
[2023-01-25 18:23:14.526] [info] [grabber] session_monitor_receiver: bound
[2023-01-25 18:23:14.526] [info] [grabber] receiver: bound
[2023-01-25 18:23:14.526] [info] [grabber] receiver: chown socket: 0
[2023-01-25 18:23:15.328] [info] [observer] grabber_client is connected.
[2023-01-25 18:23:15.328] [info] [observer] rescan devices...
[2023-01-25 18:23:18.482] [info] [observer] matching device:
[2023-01-25 18:23:18.482] [info] [observer]   registry_entry_id: <REDACTED>
[2023-01-25 18:23:18.482] [info] [observer]   product: TouchBarUserDevice
[2023-01-25 18:23:18.482] [info] [observer]   vendor_id: 1452
[2023-01-25 18:23:18.482] [info] [observer]   product_id: 34304
[2023-01-25 18:23:18.490] [info] [observer]   is_keyboard: true
[2023-01-25 18:23:18.503] [info] [observer]   is_pointing_device: false
[2023-01-25 18:23:18.503] [info] [observer] rescan devices...
[2023-01-25 18:23:18.508] [info] [observer] TouchBarUserDevice (device_id:<REDACTED>) is observed.
[2023-01-25 18:23:34.858] [info] [console_user_server] version 14.11.0
[2023-01-25 18:23:35.168] [info] [session_monitor] version 14.11.0
[2023-01-25 18:23:35.185] [info] [session_monitor] session_monitor_receiver_client is started.
[2023-01-25 18:23:35.208] [info] [session_monitor] on_console_changed: true
[2023-01-25 18:23:35.208] [info] [session_monitor] session_monitor_receiver_client is connected.
[2023-01-25 18:23:35.208] [info] [session_monitor] connected
[2023-01-25 18:23:35.210] [info] [grabber] current_console_user_id: 501
[2023-01-25 18:23:35.211] [info] [grabber] receiver is terminated
[2023-01-25 18:23:35.213] [info] [grabber] receiver is initialized
[2023-01-25 18:23:35.232] [info] [grabber] receiver: bound
[2023-01-25 18:23:35.232] [info] [grabber] receiver: chown socket: 501
[2023-01-25 18:23:35.287] [error] [observer] grabber_client error: Connection reset by peer
[2023-01-25 18:23:35.287] [info] [observer] grabber_client is closed.
[2023-01-25 18:23:36.285] [info] [observer] grabber_client is connected.
[2023-01-25 18:23:36.285] [info] [observer] rescan devices...
[2023-01-25 18:23:43.176] [info] [console_user_server] on_console_changed: on_console:true
[2023-01-25 18:23:43.179] [info] [console_user_server] grabber_client is started.
[2023-01-25 18:23:43.181] [error] [console_user_server] grabber_client connect_failed: Permission denied
[2023-01-25 18:23:44.255] [error] [console_user_server] grabber_client connect_failed: Permission denied
[2023-01-25 18:23:45.328] [error] [console_user_server] grabber_client connect_failed: Permission denied
[2023-01-25 18:23:46.409] [error] [console_user_server] grabber_client connect_failed: Permission denied
[2023-01-25 18:23:47.478] [error] [console_user_server] grabber_client connect_failed: Permission denied
[2023-01-25 18:23:48.554] [error] [console_user_server] grabber_client connect_failed: Permission denied
[2023-01-25 18:23:49.628] [error] [console_user_server] grabber_client connect_failed: Permission denied
(ad infinitum...)
```

despite

<img width="506" alt="image" src="https://user-images.githubusercontent.com/3383210/214650821-37cc39cc-5d16-4dd0-9f9b-21b9ec1eaf13.png">

What else to check?...

**Comment by heyhaiyon** [comment]
I had the same symptom on my M2 Mac mini

764308        0 drwxr-xr-x   12 root             wheel                 384  2 11 23:20 /Library/Application Support/org.pqrs/tmp
764324        0 drwx------    7 root             wheel                 224  2 11 23:20 /Library/Application Support/org.pqrs/tmp/rootonly
find: /Library/Application Support/org.pqrs/tmp/rootonly: Permission denied
764314        8 -rw-r--r--    1 root             wheel                1630  2  5 10:46 /Library/Application Support/org.pqrs/tmp/karabiner_grabber_manipulator_environment.json
764334        0 drwxr-xr-x    3 root             wheel                  96  1 14 17:30 /Library/Application Support/org.pqrs/tmp/user
764335        0 drwx------    3 hermann          wheel                  96  2  5 00:09 /Library/Application Support/org.pqrs/tmp/user/501
764336        0 drwxr-xr-x    3 502              wheel                  96  2  5 00:09 /Library/Application Support/org.pqrs/tmp/user/501/grabber_client
764337        0 srwxr-xr-x    1 502              wheel                   0  2  5 00:09 /Library/Application Support/org.pqrs/tmp/user/501/grabber_client/1740aaec53e4f470.sock
2821400        8 -rw-r--r--    1 root             wheel                  41  2 11 23:19 /Library/Application Support/org.pqrs/tmp/karabiner_grabber_state.json
2821910        0 drwxr-xr-x    3 root             wheel                  96  2 11 23:20 /Library/Application Support/org.pqrs/tmp/krbn_grabber
2821911        0 srw-------    1 hermann          wheel                   0  2 11 23:20 /Library/Application Support/org.pqrs/tmp/krbn_grabber/1742ce4ae6106248.sock
764321        0 drwxr-xr-x    5 root             wheel                 160  7 21  2021 /Library/Application Support/org.pqrs/tmp/pid
764325        8 -rw-r--r--    1 root             wheel                   5  2 11 23:20 /Library/Application Support/org.pqrs/tmp/pid/karabiner_session_monitor.501.pid
764322        8 -rw-r--r--    1 root             wheel                   5  2 11 23:19 /Library/Application Support/org.pqrs/tmp/pid/karabiner_grabber.pid
764323        8 -rw-r--r--    1 root             wheel                   5  2 11 23:19 /Library/Application Support/org.pqrs/tmp/pid/karabiner_observer.pid
764315        8 -rw-r--r--    1 root             wheel                1407  2  5 10:39 /Library/Application Support/org.pqrs/tmp/karabiner_grabber_devices.json
764312        8 -rw-r--r--    1 root             wheel                1324  2  5 10:39 /Library/Application Support/org.pqrs/tmp/karabiner_grabber_device_details.json
2821410        8 -rw-r--r--    1 root             wheel                  41  2 11 23:19 /Library/Application Support/org.pqrs/tmp/karabiner_observer_state.json
764317        8 -rw-r--r--    1 root             wheel                  18  2  5 00:09 /Library/Application Support/org.pqrs/tmp/karabiner_notification_message.json

**Comment by PawelLipski** [comment]
Reinstall helped in my case (also, IIRC the second time I've installed Karabiner via Homebrew cask)

**Comment by Zoomer547547** [comment]
This is what I did and I got it working. 
Disabled SIP
Recovery Mode > Security policy > minimized security and allowed myself to make the choice
Reboot
Uninstalled Karabiner
Reboot
Re-installed
Went to use one of my macros and then, finally, it asked me to validate the various permissions through settings. 
I think I'll experiment with it for a bit and then gradually roll back some of the security settings.
